### DISCUSSION TITLE
当el-form的disabled为true的情况下，el-form下的所有单个表单项应该有自己决定自己能否被禁用的能力。

### DISCUSSION THREAD

**Discussion by l-x-f** [element-plus/element-plus#17861] [Ideas]
下面的示例中， [【示例在线playground地址】](https://playground.plus-pro-components.com/#eNqtV91u2zYUfhXO2GAXiOWk3W6MpGgX5KJDtwbLsIvNA0pLtM2EJgWScuwFeYvdDdir7GkG7DH6HVKWJVlukrZXNnm+8/fxnEPqrvc6z5NVIXrj3qkXy1xxL15ONGOnQg1nxi4Zy6TjUyWysL0TDCXwTPGpUGeT3uvUy5X0G6b5Ukx6JTaipc4Lz1bDpckClrSTiGOjrdVR3ezjXP1pdNuVE0qke76smEuj4Q3ppWJhVCYsZLkS3AlW6mxMYdsmo1GTe6hX/n8DhgUcW3FVCGy5BdfzBZe1hD6m7G9NTXkq5LXU84ZuoCMG9hkEebl3FqlRbOxyroE7OdlLNcPxD3OZ3gi7E7A2n4SCbh3hNznlQpKmoEn5JUwzzvZhzm8UGbiVmV+M2cnx8Td1QIsZZHEwq+cgMlXcOfz3Yu2HqdAe6TRSJWwTNbd8M/zu+Bi44emI5I932UEkUf84IhHvg3zFg/wSfD2hjN5o57n2DPHKlbCbdqvdSp8u9jKqwJ/V2lRO7cpdiPRmatY4KVPkbb8thaZK1WrvtJLoPx7dSOEQJg0iKoK2Acb20TvrgdXS/ENeL61ZmjAFHum4U+HTfL+bzZ6U8j780/xeSdwlgk0t1xkT69y4wtLEPOy6W+MB743dWBlPr7qfhcPsT4maRsVZnknTXW62VNkruaCzYyE32hmaPeW/EFXAfFTvV6ELsh1+O3R2W5+ac9VpJHrMfZ0Jl+L8yjlPI5NbwZ/U5g0n08J7FHhpL7dyycPUeJUqzDxsGX1VTJfSY4+9PIcvH5mIil222Hj7VKGguXJE4f///vXf3//sa3aEW21hdTqqvYawdKmVuQd9ek7pUxc54QPzqFtjPbtjiJFYFeyezdDBrI93VZ+0RyOWGaaNZwW9N9ACoQ8YJugCWrOJTlEdPpwFO6vsDO4oKkKOWf/kuH9Ey/iWoY24Dndxc/m8tizH8ZgFPsIeUT5mv/9RmouFDJXSAR30dnX/jMKPwW3PAwEOnrGzlyxERzKjRKLMfNB3AfBVH1r3RGEkDRT1jnqRpeGS58m1MxrvzaA/KQUgdBwt0h4ePkvc2sNcFUEw6S28z914NCp0fjNPUrPEWe0wr54nL5JvRzh+P5I6E+tkViiVLK+hHZIiowQc5tbgNoRLDd2Dtjugpd2myVf1IEaSyBji0A+ZPQCvx03myTr4uwdr3gE0k/MWZxSWVMK+C4/LJndcKXP7Q9jztohHHnRoSnbsX7t1jPcStSDsCtFXMvT4XKADSXxx9ROavibEeChUmesBIU1WVVCMEfZ9oTOEXcOFaN+EEsAb+Bd3sfZCu21SFGhgI+AnPZB1/pHUd+G+SF7UWLxUfENzUmc/cqm3nztlS7faGN9DZfcmo/LbCD1c9XhAX+L4cEdTKLE4qo5PukoHhwcT3ZoD9EqtUbBofoVRODRhG/MIKXW7QVpVpGDivLAWkviQS9tzqUTuJxNBHR52ShexkEm3hNdLOwy9ODUwfOg0MTT6oUdrojTM9LdS39BIoUGkVuIIE+kaXz1HbIHB2JwzGL8RnJm0IGdJNFEGM+iTmIYPC0Bc0or8hqeyWwjh4bwUkXHI6Kfa41l2sYKdt+hGoYWFPcOzPkUUQtsZ3kcKa40NUAo+IKsgpybbJDzPhc4GpB6n40Qr4Zmko1G4sRBMvLE0v+UYsuT6igKnAtltXVxWmxNNjyScxazQuC7A8aEKiwzKGRtU7p4hUl9YGNlSG0RUJGcdlUP+YKFcUTLnBh9Va09/E1xqgz23D2rUSmiHLamIfR9YaidZIybmFRNhWtyGd7OE6XYx7coI2VZVt19y/afcBKlzsdbokjwYanVgXzzY9w9ciV/flb13X79fEPX7etS9+w/4TfmZ)
表单整体禁用，有一个"返回按钮"  自身设置了不被禁用，但是还是被禁用，此时页面无法返回，查到类似的issure
[[Component] el-form嵌套带dialog的子组件 disabled以后 dialog内层按钮都没法使用](https://github.com/element-plus/element-plus/issues/10951)，本质原因就是单个表单的`disabled`优先级低于整个`el-form`的`disabled`，换句话说单个表单的`disabled`根本没有生效。

查看源码 
https://github.com/element-plus/element-plus/blob/dev/packages/components/form/src/hooks/use-form-common-props.ts
有一段这样的代码
```ts
export const useFormDisabled = (fallback?: MaybeRef<boolean | undefined>) => {
  const disabled = useProp<boolean>('disabled')
  const form = inject(formContextKey, undefined)
  return computed(
    () => disabled.value || unref(fallback) || form?.disabled || false
  )
}
```
这里的意思是说 `disabled`取值顺序
  1. 先取组件本身props的`disabled`
  2. 然后取`fallback`参数的
  3. 再取`el-form`的
  4. 最后默认 `false`
 
但是代码中的 `||` 或，有很大的问题，就是前3个任意一个返回`true` 整个函数就返回`true`，那上面的取值顺序就变得没有意义了。

理想的做法应该是  把 `||` 换成 `??`

```ts
  () => disabled.value ?? unref(fallback) ?? form?.disabled ?? false
```

这样基本可以解决上面的问题，即能保证取值也能保证顺序（优先级）。

另外查过ant.design 代码 [【链接】](https://github.com/ant-design/ant-design/blob/master/components/button/button.tsx)，发现它是可以正常实现上面功能的。

**Comment by LeoZeda** [comment]
遇到这个问题，有更新吗？现在用内嵌一个el-form 解决
